packages/medusa/src/api-v2/admin/draft-orders/validators.ts

```diff
@@ -0,0 +1,181 @@
+import { BigNumberInput } from "@medusajs/types"
+import { Type } from "class-transformer"
+import {
+  IsArray,
+  IsBoolean,
+  IsEmail,
+  IsEnum,
+  IsNotEmpty,
+  IsNumber,
+  IsObject,
+  IsOptional,
+  IsString,
+  ValidateIf,
+  ValidateNested,
+} from "class-validator"
+import {
+  AddressPayload,
+  FindParams,
+  extendedFindParamsMixin,
+} from "../../../types/common"
+import { IsType } from "../../../utils/validators/is-type"
+
+export class AdminGetOrdersOrderParams extends FindParams {}
+/**
+ * Parameters used to filter and configure the pagination of the retrieved api keys.
+ */
+export class AdminGetOrdersParams extends extendedFindParamsMixin({
+  limit: 50,
+  offset: 0,
+}) {
+  /**
+   * Search parameter for api keys.
+   */
+  @IsString({ each: true })
+  @IsOptional()
+  id?: string | string[]
+
+  /**
+   * Filter by title
+   */
+  @IsString({ each: true })
+  @IsOptional()
+  name?: string | string[]
+
+  // Additional filters from BaseFilterable
+  @IsOptional()
+  @ValidateNested({ each: true })
+  @Type(() => AdminGetOrdersParams)
+  $and?: AdminGetOrdersParams[]
+
+  @IsOptional()
+  @ValidateNested({ each: true })
+  @Type(() => AdminGetOrdersParams)
+  $or?: AdminGetOrdersParams[]
```

Comment:
Q: do we really want that as part of the http filters? eventually OperatorMap on id and name?
If we really want to provide that, we need to rework it a bit I believe, at the moment each item of $and and $or would include the pagination